dspback/routers/earthchem.py

```diff
@@ -158,7 +161,19 @@ async def delete_metadata_repository(self, request: Request, identifier):
         description="Creates a submission record of the EarthChem record.",
     )
     async def submit_repository_record(self, request: Request, identifier: str):
-        json_metadata = await self.submit(request, identifier)
+        try:
```

Comment:
If Peng can update the ECL-API to allow a GET for published records when a user is not the owner then this change block can be reverted.